packages/@ama-sdk/schematics/schematics/typescript/core/openapi-codegen-typescript/src/main/resources/typescriptFetch/api/api.mustache

```diff
@@ -65,7 +65,7 @@ export class {{classname}} implements Api {
   {{/isDeprecated}}
    * @param data Data to provide to the API call
    */
-  public async {{nickname}}(data: {{#uppercaseFirst}}{{nickname}}{{/uppercaseFirst}}RequestData, metadata?: RequestMetadata<{{#consumes}}'{{mediaType}}'{{^-last}} | {{/-last}}{{/consumes}}{{^consumes}}string{{/consumes}}, {{#produces}}'{{mediaType}}'{{^-last}} | {{/-last}}{{/produces}}{{^produces}}string{{/produces}}>): Promise<{{#vendorExtensions}}{{#responses2xx}}{{dataType}}{{^dataType}}void{{/dataType}}{{^-last}} | {{/-last}}{{/responses2xx}}{{^responses2xx}}void{{/responses2xx}}{{/vendorExtensions}}> {
+  public async {{nickname}}(data: {{#uppercaseFirst}}{{nickname}}{{/uppercaseFirst}}RequestData, metadata?: RequestMetadata<{{#consumes}}'{{mediaType}}'{{^-last}} | {{/-last}}{{/consumes}}{{^consumes}}string{{/consumes}}, {{#produces}}'{{mediaType}}'{{^-last}} | {{/-last}}{{/produces}}{{^produces}}string{{/produces}}>): Promise<{{#vendorExtensions}}{{#responses2xxReturnTypes}}{{.}}{{^-last}} | {{/-last}}{{/responses2xxReturnTypes}}{{^responses2xxReturnTypes}}void{{/responses2xxReturnTypes}}{{/vendorExtensions}}> {
```

Comment:
won't that be a breaking change however? :D

Reply:
Actually I don't think we have the case at all but is better to have a breaking change `void` -> `never` or to let people hope to have a promise resolve that will never happen :D?

(I even don't no if `Promise<never>` is valid btw :)

Reply:
Well, I re-read the documentation and never would mean that it does not return, that the method does not end. In our case, the Http call will always send a response whether there is or not a body. It will end, and the lack of response is most likely an issue on the specification itself

Reply:
Not sure I get the answer :S.
Actually the http call resolves but the the function should not right?
We are in the case that, in the spec, only status different of 2xx are defined which me that we are in case we `response.ok == true` which will always reject the function (so it does not resolve).
According to https://swagger.io/docs/specification/describing-responses/ it is not "common" to not have success response but not prohibited.
According to that it really sounds that `Promise<never>` as return type would make sense to indicate that the function never resolve (because it only reject) and it is not really a case that we can ignored as it does not seem to be prohibited by the spec :S.